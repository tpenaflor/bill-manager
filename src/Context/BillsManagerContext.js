import React from 'react'

const BillsManagerContext = React.createContext()

export default BillsManagerContext