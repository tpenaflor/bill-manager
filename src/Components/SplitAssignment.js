import React, {useContext} from 'react'
import BillAssignment from './BillAssigment'
import BillsManagerContext from '../Context/BillsManagerContext'

const SplitAssignment = ({bill}) => {
    const {participants} = useContext(BillsManagerContext)
    return participants.map((part) => (
        <BillAssignment key={part.id} part={part} bill={bill}/>
    ))
}


export default SplitAssignment