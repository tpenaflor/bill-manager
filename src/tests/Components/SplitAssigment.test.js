import React from 'react'
import {shallow} from 'enzyme'
import { shallowToJson  } from 'enzyme-to-json';

import SplitAssignment from '../../Components/SplitAssignment'
import * as BillsManagerContext from '../../Context/BillsManagerContext'
import {participantFixture} from '../Fixtures/ParticipantFixture'

test('render bill list', ()=> {
    const contextValues = {
        participants : participantFixture
    }

    jest.spyOn(BillsManagerContext, 'useBillsManagerContext')
        .mockImplementation(() => contextValues)

    const wrapper = shallow(<SplitAssignment />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
})