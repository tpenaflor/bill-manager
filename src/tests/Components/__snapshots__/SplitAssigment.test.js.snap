// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render bill list 1`] = `
Array [
  <BillAssignment
    key="a4714460-a11c-4681-a21d-c7de2c57a02c"
    part={
      Object {
        "id": "a4714460-a11c-4681-a21d-c7de2c57a02c",
        "name": "user1",
      }
    }
  />,
  <BillAssignment
    key="ac33c426-53c4-4d38-ba0f-f8e569a9a507"
    part={
      Object {
        "id": "ac33c426-53c4-4d38-ba0f-f8e569a9a507",
        "name": "user2",
      }
    }
  />,
  <BillAssignment
    key="3712fdf0-5b62-41b6-9442-b35a41d5ec81"
    part={
      Object {
        "id": "3712fdf0-5b62-41b6-9442-b35a41d5ec81",
        "name": "user3",
      }
    }
  />,
  <BillAssignment
    key="6be42a94-61eb-47d4-8552-0369487f015c"
    part={
      Object {
        "id": "6be42a94-61eb-47d4-8552-0369487f015c",
        "name": "user4",
      }
    }
  />,
]
`;
